src/main/java/de/tum/in/www1/artemis/service/hestia/ExerciseHintService.java

```diff
@@ -255,4 +257,25 @@ private Set<ExerciseHint> getAvailableExerciseHintsForTask(Optional<Integer> sub
         }
         return availableHintsForTask;
     }
+
+    /**
+     * Returns the title of the hint identified by the given hint id if the exercise id stored in the hint matches the
+     * provided exercise id.
+     *
+     * @param exerciseId the exercise id that must match the one stored in the hint
+     * @param exerciseHintId the id of the hint
+     * @return the title of the hint if it was found; null otherwise
+     *
+     * @throws ConflictException if the provided exercise id does not match the one stored in the hint
+     */
+    @Cacheable(cacheNames = "exerciseHintTitle", key = "#{exerciseId + '-' + exerciseHintId}", unless = "#result == null")
+    public String getExerciseHintTitle(Long exerciseId, Long exerciseHintId) {
+        final var hint = exerciseHintRepository.findByIdElseThrow(exerciseHintId);
+
+        if (hint.getExercise() == null || !hint.getExercise().getId().equals(exerciseId)) {
+            throw new ConflictException("An exercise hint can only be retrieved if the exerciseIds match.", "exerciseHint", "exerciseIdsMismatch");
+        }
+
+        return hint.getTitle() == null ? null : hint.getTitle();
```

Comment:
```suggestion
        return hint.getTitle();
```
I don't see any semantical difference to my suggestion

Reply:
Thanks, solved in 3e828ca 👍